components/viaduct/src/backend/reqwest.rs

```diff
@@ -0,0 +1,96 @@
+            .connect_timeout(GLOBAL_SETTINGS.connect_timeout)
+            .redirect(
+                if GLOBAL_SETTINGS.follow_redirects {
+                    reqwest::RedirectPolicy::default()
+                } else {
+                    reqwest::RedirectPolicy::none()
+                }
+            )
+            // Note: no cookie or cache support.
+            .build()
+            .expect("Failed to initialize global reqwest::Client")
+    };
+}
+
+// Implementing From to do this would end up being public
+impl crate::Request {
+    fn into_reqwest(self) -> Result<reqwest::Request, crate::Error> {
+        let method = match self.method {
+            crate::Method::Get => reqwest::Method::GET,
+            crate::Method::Head => reqwest::Method::HEAD,
+            crate::Method::Post => reqwest::Method::POST,
+            crate::Method::Put => reqwest::Method::PUT,
+            crate::Method::Delete => reqwest::Method::DELETE,
+            crate::Method::Connect => reqwest::Method::CONNECT,
+            crate::Method::Options => reqwest::Method::OPTIONS,
+            crate::Method::Trace => reqwest::Method::TRACE,
+        };
+        let mut result = reqwest::Request::new(method, self.url);
+        for h in self.headers {
+            use reqwest::header::{HeaderName, HeaderValue};
+            // Unwraps should be fine, we verify these in `Header`
+            let value = HeaderValue::from_str(&h.value).unwrap();
+            result
+                .headers_mut()
+                .insert(HeaderName::from_bytes(h.name.as_bytes()).unwrap(), value);
+        }
+        *result.body_mut() = self.body.map(reqwest::Body::from);
+        Ok(result)
+    }
+}
+
+pub fn send(request: crate::Request) -> Result<crate::Response, crate::Error> {
+    let request_method = request.method;
+    let req = request.into_reqwest()?;
+    let mut resp = CLIENT.execute(req).map_err(|e| {
+        log::error!("Reqwest error: {:?}", e);
+        crate::Error::NetworkError(e.to_string())
+    })?;
+    let status = resp.status().as_u16();
+    let url = resp.url().clone();
+    let mut body = Vec::with_capacity(resp.content_length().unwrap_or_default() as usize);
+    resp.read_to_end(&mut body).map_err(|e| {
+        log::error!("Failed to get body from response: {:?}", e);
+        crate::Error::NetworkError(e.to_string())
+    })?;
+    let mut headers = crate::Headers::with_capacity(resp.headers().len());
+    for (k, v) in resp.headers() {
+        let val = String::from_utf8_lossy(v.as_bytes()).to_string();
+        let hname = match crate::HeaderName::new(k.as_str().to_owned()) {
+            Ok(name) => name,
+            Err(e) => {
+                // Ignore headers with invalid names, since nobody can look for them anyway.
+                log::warn!("Server sent back invalid header name: '{}'", e);
+                continue;
+            }
+        };
+        // Not using Header::new intentionally, since the error it returns is
```

Comment:
Worth having a `new_unchecked` or `new_from_response` (and `new_for_request`) method that either creates the header as-is, or returns `Option<Header>`?